### PR TITLE
Python: Add Logit-Bias to TextCompletion abstractions

### LINKED ISSUE #1854: Python: Add Logic-Bias and any potential property missing to TextCompletion abstractions.

### PR DESCRIPTION
**Pull request by SebasX5**
### Motivation and Context

1. Adds the Logit-Bias feature that is in the C# into Python.
2. It adds the token bias configuration to the text and chat completion settings.
3. It contributes to flexibility in prompt engineering by allowing the user to influence the likelihood of specified tokens appearing in the text and chat completion.
4. Closes #1854 

### Description

We added a dictionary field called token_selection_biases to the classes CompleteRequestSettings and ChatRequestSettings to implement logit bias and included an example to demonstrate functionality.
### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: python/tests/unit/ai/open_ai/services/test_azure_chat_completion.py [hunk 1]
```diff
@@ -154,4 +154,5 @@ async def test_azure_chat_completion_call_with_parameters() -> None:
             frequency_penalty=complete_request_settings.frequency_penalty,
             n=complete_request_settings.number_of_responses,
             stream=False,
```
Comment (awharrison-28) [inline python/tests/unit/ai/open_ai/services/test_azure_chat_completion.py:156]: Please add a unit test demonstrating a call with logit_bias != None

File: python/tests/unit/ai/open_ai/services/test_azure_text_completion.py [hunk 2]
```diff
@@ -153,4 +153,5 @@ async def test_azure_text_completion_call_with_parameters() -> None:
             stop=None,
             n=complete_request_settings.number_of_responses,
             stream=False,
```
Comment (awharrison-28) [inline python/tests/unit/ai/open_ai/services/test_azure_text_completion.py:155]: Please add a unit test demonstrating a call with logit_bias != None

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py [hunk 3]
```diff
@@ -101,17 +101,8 @@ async def complete_async(
             str -- The completed text.
         """
         prompt_to_message = [("user", prompt)]
-        chat_settings = ChatRequestSettings(
```
Comment (awharrison-28) [inline python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py:104]: What's the rationale for removing the ChatRequestSettings creation here? This method and `complete_stream_async` are here so that a chat model can be used as a few-shot text completion model like text-davinci-003. 

I agree that it's a bit wasteful to have the code here. It could be abstracted to `_send_chat_request` such that  `_send_chat_request` can accept either ChatRequestSettings or CompleteRequestSettings and does the conversion there. 

I could totally be missing something too.

**Review by awharrison-28 — CHANGES_REQUESTED** [review]
Love the syntax example

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py (hunk 3 above)
Comment (sneha-afk) [reply python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py:104]: We realized that there is a difference between `complete..` and `complete_chat`, we will be adding it back soon.

Comment (sneha-afk) [reply python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py]: I have added back the original `ChatRequestSettings` creation, now with also addressing the `token_selection_biases` field. Thank you!

File: python/samples/kernel-syntax-examples/logit_bias.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
+# Copyright (c) Microsoft. All rights reserved.
```
Comment (alexchaomander) [inline python/samples/kernel-syntax-examples/logit_bias.py:1]: I would consider naming this example as openai_logit_bias.py so it's clear that this is for the OpenAI API. Other models might have something slightly different.

**Review by alexchaomander — DISMISSED** [review]
Looks good to me! Great example. Left one comment to consider

**Review by awharrison-28 — APPROVED** [review]
